Patch by MSP-greg
allows RubyGems tests to be run from install folder
diff --git a/lib/rubygems/installer.rb b/lib/rubygems/installer.rb
index 2ba92482cc..d4ec1d613e 100644
--- a/lib/rubygems/installer.rb
+++ b/lib/rubygems/installer.rb
@@ -715,10 +715,13 @@ def process_options # :nodoc:
 
   def check_that_user_bin_dir_is_in_path # :nodoc:
     user_bin_dir = @bin_dir || Gem.bindir(gem_home)
-    user_bin_dir = user_bin_dir.gsub(File::SEPARATOR, File::ALT_SEPARATOR) if
-      File::ALT_SEPARATOR
 
     path = ENV['PATH']
+    if File::ALT_SEPARATOR
+      user_bin_dir = user_bin_dir.gsub(File::SEPARATOR, File::ALT_SEPARATOR)
+      path = path.gsub(File::SEPARATOR, File::ALT_SEPARATOR)
+    end
+
     if Gem.win_platform?
       path = path.downcase
       user_bin_dir = user_bin_dir.downcase
